pennylane/ops/qubit/arithmetic_ops.py

```diff
@@ -305,3 +310,257 @@ def compute_decomposition(wires):
+        if wires is None:
+            raise ValueError("Must specify wires that the operation acts on.")
+        if len(wires) > 1:
+            control_wires = Wires(wires[:-1])
+            wires = Wires(wires[-1])
+        else:
+            raise ValueError(
+                "IntegerComparator: wrong number of wires. "
+                f"{len(wires)} wire(s) given. Need at least 2."
+            )
+
+        total_wires = control_wires + wires
+
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["target_wires"] = wires
+        self.hyperparameters["value"] = value
+        self.hyperparameters["geq"] = geq
+        self.geq = geq
+        self.value = value
+
+        super().__init__(wires=total_wires, do_queue=do_queue)
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        return base_label or f">={self.value}" if self.geq else f"<{self.value}"
+
+    # pylint: disable=unused-argument
+    @staticmethod
+    def compute_matrix(
+        value=None, control_wires=None, geq=True, **kwargs
+    ):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.IntegerComparator.matrix`
+
+        Args:
+            value (int): The value :math:`L` that the state's decimal representation is compared against.
+            control_wires (Union[Wires, Sequence[int], or int]): wires to place controls on
+            geq (bool): If set to `True`, the comparison made will be :math:`n \geq L`. If `False`, the comparison
+                made will be :math:`n < L`.
+
+        Returns:
+           tensor_like: matrix representation
+
+        **Example**
+
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1]))
+        [[1. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 1. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 1. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 1. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 1. 0. 0.]
+         [0. 0. 0. 0. 1. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 1.]
+         [0. 0. 0. 0. 0. 0. 1. 0.]]
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1], geq=False))
+        [[0. 1. 0. 0. 0. 0. 0. 0.]
+         [1. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 1. 0. 0. 0. 0.]
+         [0. 0. 1. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 1. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 1. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 1. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 1.]]
+        """
+
+        if value is None:
+            raise ValueError("The value to compare to must be specified.")
+        if control_wires is None:
+            raise ValueError("Must specify the control wires.")
+        if isinstance(value, int):
+            binary = "0" + str(len(control_wires)) + "b"
```

Comment:
Minor: there are two branches out of 4 where `binary` is not being used. A way for refactoring the logic could be to consider the edge case of creating an identity matrix first, e.g., by
```py
    small_val = not geq and value == 0
    large_val = geq and value > 2 ** len(control_wires) - 1
    if small_val or large_val:
        return np.eye(2 ** (len(control_wires) + 1))
```
This also helps consider two identical cases at once and gets this part of the logic "out of the way" before the heavier lifting is done.